nav2_ceres_costaware_smoother/include/nav2_ceres_costaware_smoother/options.hpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2021 RoboTech Vision
+// Copyright (c) 2020, Samsung Research America
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. Reserved.
+
+#ifndef NAV2_CERES_COSTAWARE_SMOOTHER__OPTIONS_HPP_
+#define NAV2_CERES_COSTAWARE_SMOOTHER__OPTIONS_HPP_
+
+#include <string>
+#include <vector>
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "nav2_util/node_utils.hpp"
+
+namespace nav2_ceres_costaware_smoother
+{
+
+/**
+ * @struct nav2_smac_planner::SmootherParams
+ * @brief Parameters for the smoother cost function
+ */
+struct SmootherParams
+{
+  /**
+   * @brief A constructor for nav2_smac_planner::SmootherParams
+   */
+  SmootherParams()
+  {
+  }
+
+  /**
+   * @brief Get params from ROS parameter
+   * @param node_ Ptr to node
+   * @param name Name of plugin
+   */
+  void get(rclcpp_lifecycle::LifecycleNode * node, const std::string & name)
+  {
+    std::string local_name = name + std::string(".");
+
+    // Smoother params
+    double minimum_turning_radius;
+    nav2_util::declare_parameter_if_not_declared(
+      node, name + ".minimum_turning_radius", rclcpp::ParameterValue(0.4));
+    node->get_parameter(name + ".minimum_turning_radius", minimum_turning_radius);
+    max_curvature = 1.0f / minimum_turning_radius;
+    nav2_util::declare_parameter_if_not_declared(
+      node, local_name + "w_curve", rclcpp::ParameterValue(30.0));
+    node->get_parameter(local_name + "w_curve", curvature_weight);
+    nav2_util::declare_parameter_if_not_declared(
+      node, local_name + "w_cost", rclcpp::ParameterValue(0.015));
+    node->get_parameter(local_name + "w_cost", costmap_weight);
+    nav2_util::declare_parameter_if_not_declared(
+      node, local_name + "w_cost_cusp", rclcpp::ParameterValue(0.04));
```

Comment:
Can you explain this weight a bit, why its needed, its effect, and where it is applied?

Reply:
Let the [tests](https://github.com/robotechvision/navigation2/blob/bd4ec4c87eb0c4383a35736171c0841b4dda84f9/nav2_ceres_costaware_smoother/test/test_costaware_smoother.cpp#L721) speak for me here. Since the cusps are usually surrounded by rotations, the optimizer tends to stick in a local minima, not moving the cusp point away from obstacles. A higher cost value is needed to help him out.

This picture (cyan - input path, purple - path smoothed with no cusp zone/cost, green - path smoothed with a higher cusp cost) shows how wobbliness is eliminated and cusp is pushed into a safer place when `w_cost_cusp` and `cusp_zone_length` are used.
![image](https://user-images.githubusercontent.com/6342166/160766407-63272f0f-ae8d-4256-b052-687eb9a41c2a.png)

Reply:
Got it, this would be good graphic / explanation to add to the readme of the package in explaining those parameters. I figured as much was true, but this is a  different way than I would have gone about it - works though.

Reply:
Updated the picture with shorter arrow lengths, so now there should be less clutter. The purple path is continuous, but apparently worse than the green one (note: `w_curve` was set to a very low value here to make other parameters show up, so turning radius is not relevant here).

Reply:
Got it - OK I like this behavior and this makes sense.